classes/submissionFile/DAO.inc.php

```diff
@@ -0,0 +1,465 @@
+            ->exists();
+
+        if ($hasSubmissionFileRevision) {
+            return;
+        }
+
+        DB::table('submission_file_revisions')->insert([
+            'submission_file_id' => $submissionFile->getId(),
+            'file_id' => $submissionFile->getData('fileId'),
+        ]);
+    }
+
+    /**
+     * @copydoc EntityDAO::delete()
+     */
+    public function delete(SubmissionFile $submissionFile)
+    {
+        parent::_delete($submissionFile);
+    }
+
+    /**
+     * @copydoc EntityDao::deleteById()
+     */
+    public function deleteById(int $submissionFileId)
+    {
+        DB::table('submission_file_revisions')
+            ->where('submission_file_id', '=', $submissionFileId)
+            ->delete();
+
+        DB::table('review_round_files')
+            ->where('submission_file_id', '=', $submissionFileId)
+            ->delete();
+
+        $reviewFilesDAO = DAORegistry::getDAO('ReviewFilesDAO');
+        $reviewFilesDAO->revokeBySubmissionFileId($submissionFileId);
+
+        parent::deleteById($submissionFileId);
+    }
+
+    /**
+     * Get the files for each revision of a submission file
+     */
+    public function getRevisions(int $submissionFileId): Collection
+    {
+        return DB::table('submission_file_revisions as sfr')
+            ->leftJoin('files as f', 'f.file_id', '=', 'sfr.file_id')
+            ->where('submission_file_id', '=', $submissionFileId)
+            ->orderBy('revision_id', 'desc')
+            ->select(['f.file_id as fileId', 'f.path', 'f.mimetype'])
+            ->get();
+    }
+
+    /**
+     * Retrieve file by public file ID
+     *
+     * $pubIdType it is one of the NLM pub-id-type values or
+     * 'other::something' if not part of the officia.l NLM list
+     * (see <http://dtd.nlm.nih.gov/publishing/tag-library/n-4zh0.html>).
+     *
+     * @param null|mixed $submissionId
+     * @param null|mixed $contextId
+     */
+    public function getByPubId(
+        $pubIdType,
+        $pubId,
+        $submissionId = null,
+        $contextId = null
+    ): ?SubmissionFile {
+        if (empty($pubId)) {
+            return null;
+        }
+
+        $submissionFileId = DB::table('submission_files as sf')
+            ->leftJoin($this->settingsTable . ' as sfs', 'sfs.submission_file_id', '=', 'sf.submission_file_id')
+            ->where('sf.submission_id', '=', $submissionId)
+            ->where(function ($q) use ($pubIdType, $pubId) {
+                $q->where('sfs.setting_name', '=', 'pub-id::' . $pubIdType);
+                $q->where('sfs.setting_value', '=', $pubId);
+            })
+            ->select('sf.*')
+            ->groupBy('sf.submission_file_id')
+            ->value('sf.submission_file_id');
+
+        if (empty($submissionFileId)) {
+            return null;
+        }
+
+        $submissionFile = $this->get($submissionFileId);
+
+        if (!$submissionFile->itsOnFileProofStage()) {
+            return $submissionFile;
+        }
```

Comment:
Is the logic reversed here? I think it should return the file if it's in the proof file stage.

Reply:
Currently this is the way we did it. Even on [main branch](https://github.com/pkp/pkp-lib/blob/bf272b821ce4d5d13cc2645426a02722663bcc59/classes/submission/PKPSubmissionFileDAO.inc.php#L228).

Reply:
Of course!

Reply:
I think only `SubmissionFile::SUBMISSION_FILE_PROOF` for this one. (dependent files can't have pub ids)